smarkets/streaming_api/framing.py

```diff
@@ -23,16 +23,16 @@ def frame_decode_all(string):
     :rtype: tuple of (list of payloads) and remaining string
     """
     payloads = []
-    frame_size = 4
-    while len(string) >= frame_size:
+    min_frame_size = 4
+    while len(string) >= min_frame_size:
         try:
             decoded = uleb128_decode(string)
         except IncompleteULEB128:
             # There may be not enough data in the input string to decode the header
             break
         else:
             payload_size, header_size = decoded
-            frame_size = max(payload_size + header_size, 4)
+            frame_size = max(payload_size + header_size, min_frame_size)
             if len(string) >= frame_size:
                 frame, string = string[:frame_size], string[frame_size:]
                 payloads.append(frame[header_size:header_size + payload_size])
```

Comment:
Hmmm... unrelated to this PR, I think there is a lurking disastrous bug here: if we don't enter the ` if len(string) >= frame_size` thing, then `string` will not change, so we will enter the `while` again and loop forever....